cartoframes/data/observatory/category.py

```diff
@@ -22,7 +23,13 @@ def get_by_id(category_id):
         return get_category_repo().get_by_id(category_id)
 
     def datasets(self):
-        return get_dataset_repo().get_by_category(self[_CATEGORY_ID_FIELD])
+        return get_dataset_repo().get_by_category(self.get_id())
+
+    def get_id(self):
```

Comment:
Should be a "private" method? I think it is nor part of the public API, so I would change it to `_get_id`. Same in the rest of classes

Reply:
Yes, I had the same doubt... It could be useful for the final user but we put the index for that so I'll make it private :dancer: